extensions/amp-fit-text/1.0/amp-fit-text.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {FitText} from './fit-text';
+import {PreactBaseElement} from '../../../src/preact/base-element';
+import {getLengthNumeral, isLayoutSizeDefined} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+
+import {dict} from '../../../src/utils/object';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+const TAG = 'amp-fit-text';
+
+const getFontSizeAttrs = (element) =>
+  dict({
+    'maxFontSize': getLengthNumeral(element.getAttribute('max-font-size')),
+    'minFontSize': getLengthNumeral(element.getAttribute('min-font-size')),
+  });
+
+class AmpFitText extends PreactBaseElement {
+  /** @override */
+  init() {
+    const attributeOb = new MutationObserver(() => {
+      this.mutateProps(getFontSizeAttrs(this.element));
+    });
+    attributeOb.observe(this.element, {
+      attributeFilter: ['min-font-size', 'max-font-size'],
+      attributes: true,
+    });
+
+    // Force render to resize to new contents.
+    const childOb = new MutationObserver(() => {
+      this.mutateProps(dict({}));
+    });
+    childOb.observe(this.element, {
+      childList: true,
+      subtree: true,
+      attributes: true,
+    });
+
+    const props = getFontSizeAttrs(this.element);
+    props['width'] = this.element./* OK */ offsetWidth;
+    props['height'] = this.element./* OK */ offsetHeight;
```

Comment:
@dvoytenko @jridgewell FYI since I know we prefer to use `AmpFoo['props']` whenever possible -- I discovered from the e2e tests that the font size should be sized to the offset dimensions of the element if we are to stay consistent with 0.1

Reply:
I think sizing to offset dimensions was always a little bit wrong. The right dimension are `clientWidth/Height`. But we can make it so it doesn't matter.

Reply:
On one hand, we are saying "the dev should specify `width`/`height` via `styles`" and on the other we are saying we should not explicitly pass dimensions from AMP mode. In this case we are the dev who should specify the dimensions. I'm OK with doing this via `prop.style` instead of via `prop` but I don't think we can get away with doing neither.

Reply:
Certainly. AMP would just pass `style={width:100%, height:100%}`, right?

Reply:
The only thing this changes is the e2e test case wherein the expected font size accounts for the border dimensions. Are we OK with breaking that?